buildenv/jenkins/jobs/infrastructure/Cleanup-Nodes.groovy

```diff
@@ -163,6 +163,9 @@ timeout(time: TIMEOUT_TIME.toInteger(), unit: TIMEOUT_UNITS) {
                                         cleanDirsStr += cleanDirs.join(" ${buildWorkspace}/../../")
                                         // shared classes cache
                                         cleanDirsStr += " ${buildWorkspace}/../../javasharedresources /tmp/javasharedresources /temp/javasharedresources"
+                                        //to remove {buildWorkspace}/../../test* directories except testDependency
+                                        cleanDirsStr = cleanDirsStr.replaceAll("${buildWorkspace}/../../test\\*", "")
+                                        sh(script: "find ${buildWorkspace}/../../ -maxdepth 1 -name 'TestDependency*' -prune -o -name 'test*' -exec rm -rf {} +", returnStdout: true).trim()
```

Comment:
I think you mean `testDependency`; lowercase `t`, without the wildcard.
Please also add a space to start the comment:
```
                                        // to remove {buildWorkspace}/../../test* directories except testDependency
```
A simpler, perhaps clearer solution is
```
    sh(script: "ls ${buildWorkspace}/../../test* | grep -v '/testDependency$' | xargs rm -rf", returnStdout: true).trim()
```